be/src/vec/runtime/shared_hash_table_controller.cpp

```diff
@@ -22,23 +22,24 @@
 namespace doris {
 namespace vectorized {
 
-bool SharedHashTableController::should_build_hash_table(RuntimeState* state, int my_node_id) {
+bool SharedHashTableController::should_build_hash_table(const TUniqueId& fragment_instance_id,
+                                                        int my_node_id) {
     std::lock_guard<std::mutex> lock(_mutex);
     auto it = _builder_fragment_ids.find(my_node_id);
     if (it == _builder_fragment_ids.cend()) {
```

Comment:
If the builder instance's hash join node closed, should remove its id from the builder fragments and signal all waiters.
In wait for hash table method,  should check if the builder fragments still exists in case of the builder fragments exist during errors.